echopop/nwfsc_feat/get_proportions.py

```diff
@@ -1,162 +1,1529 @@
+    >>> female_large_props = get_number_proportions_slice(
+    ...     aged_data,
+    ...     exclude_filter={"length_bin": small_length_bins},
+    ...     include_filter={"sex": ["female"]}
+    ... )
+    """
+
+    # Determine index columns from filter keys and length_bin
+    index_set = set(list(exclude_filter.keys()) + list(include_filter.keys()) + ["length_bin"])
+
+    # Intersect with actually available columns/indices
+    available_columns = set(list(number_proportions.index.names) + list(number_proportions.columns))
+    index_set = index_set.intersection(available_columns)
+
+    # Separate stratification columns from index columns
+    column_set = set(stratify_by).difference(index_set)
+
+    # Create pivot table
+    group_table = utils.create_pivot_table(
+        number_proportions,
+        index_cols=list(index_set),
+        strat_cols=list(column_set),
+        value_col="proportion",
+    )
+
+    # Apply filters
+    filtered_table = utils.apply_filters(
+        group_table, include_filter=include_filter, exclude_filter=exclude_filter
+    )
+
+    # Handle aggregation if no filter is applied
+    if len(include_filter) == 0 and len(exclude_filter) == 0:
+        return filtered_table
+
+    # Handle aggregation based on stratification complexity
+    if len(stratify_by) == 1:
+        # Single stratification variable - sum over all other dimensions
+        aggregated_result = filtered_table.unstack("length_bin").sum().unstack(stratify_by)
+        return aggregated_result.sum(axis=0)
+    else:
+        # Multiple stratification variables - preserve structure
+        current_dimensions = set(
+            list(filtered_table.index.names) + list(filtered_table.columns.names)
+        )
+
+        # If already at target dimensions, return as-is
+        if len(current_dimensions.difference(set(stratify_by))) == 0:
+            return filtered_table
+        else:
+            # Aggregate step by step
+            intermediate_result = filtered_table.unstack("length_bin").sum()
+
+            # Find additional dimensions to unstack
+            remaining_dimensions = list(
+                set(stratify_by).difference(intermediate_result.index.names)
+            )
+
+            # Fill in missing dimensions if needed
+            if len(remaining_dimensions) == 0:
+                remaining_dimensions = list(set(stratify_by).difference(["length_bin"]))
+
+            # Final unstack
+            return intermediate_result.unstack(remaining_dimensions)
+
+
+def get_weight_proportions_slice(
+    weight_proportions: pd.DataFrame,
+    stratify_by: List[str],
+    include_filter: Dict[str, Any] = {},
+    number_proportions: Optional[Union[Dict[str, pd.DataFrame], pd.DataFrame]] = None,
+    length_threshold_min: float = 0.0,
+    weight_proportion_threshold: float = 1e-10,
+) -> pd.Series:
+    """
+    Calculate weight proportions for a population slice with optional thresholding.
```

Comment:
Making the description consistent.
```suggestion
    Calculate weight proportions for a specific slice of the population with optional thresholding.
```